Externals.cfg

```diff
@@ -29,7 +29,7 @@ protocol = git
 required = True
 
 [GSI-EnKF]
-hash = 48d8676
+hash = 31b8b29
```

Comment:
This hash is fine but FYI, NOAA-EMC/GSI develop has new commits.  The current head is [`89ca542`](https://github.com/NOAA-EMC/GSI/commit/89ca542c963c7887494e728656515918ce7a8518).  Five open PRs will be likely merged into develop over the coming weeks.

Reply:
@RussTreadon-NOAA Noted, thanks! I'll let those other incoming PRs update the GSI hash further.